tomcat/build.go

```diff
@@ -151,6 +154,11 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 		libcnb.Process{Type: "web", Command: command, Arguments: arguments, Direct: true, Default: true},
 	)
 
+	bomScanner := sbom.NewSyftCLISBOMScanner(context.Layers, effect.NewExecutor(), b.Logger)
+	if err := bomScanner.ScanBuild(context.Application.Path, libcnb.CycloneDXJSON, libcnb.SyftJSON); err != nil {
+		return libcnb.BuildResult{}, fmt.Errorf("unable to create Build SBoM \n%w", err)
+	}
+
```

Comment:
This is actually running the Syft scanner when tests run. It's more code, but we need to have the code using the SBOMScanner interface, so that a mock implementation can be passed in for testing. We don't want to require `syft` be installed to run the tests.

Normally, you'd inject the SBOMScanner when you create Build, but the SBOMScanner needs variables only available during build to create the Syft Scanner. I don't know of a great solution but I encountered a similar solution in the Rust Cargo buildpack and what I did there was to add the scanner interface to the build struct, but and do a nil check during build. If it's nil, then create it. If it's not nill, then use it. That way during tests, you can inject a mock scanner. There is a mock scanner in libpak.

https://github.com/paketo-community/cargo/blob/main/cargo/build.go#L63-L71